ui/src/alerting/components/notifications/NewRuleOverlay.tsx

```diff
@@ -1,27 +1,95 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, useReducer, Dispatch} from 'react'
 import {withRouter, WithRouterProps} from 'react-router'
 
 // Components
-import {Overlay} from '@influxdata/clockface'
+import RuleSchedule from 'src/alerting/components/notifications/RuleSchedule'
+import RuleConditions from 'src/alerting/components/notifications/RuleConditions'
+import RuleMessage from 'src/alerting/components/notifications/RuleMessage'
+import {
+  Panel,
+  ComponentSize,
+  Overlay,
+  Form,
+  Input,
+  Grid,
+  Columns,
+} from '@influxdata/clockface'
+
+// Reducers
+import {reducer, RuleState, Actions} from './NewRuleOverlay.reducer'
+
+// Constants
+import {newRule, endpoints} from 'src/alerting/constants'
+
+// Types
+import {NotificationRuleBox} from 'src/types'
 
 type Props = WithRouterProps
 
+export const newRuleState: RuleState = {
+  ...newRule,
+  schedule: 'every',
+}
+
+export const NewRuleDispatch = React.createContext<Dispatch<Actions>>(null)
```

Comment:
I've been digging this pattern a lot. I was thinking we could codify it into a little utility. Something like:

```
const [MyStateProvider, useMyState, useMyDispatch] = createReducerContext(reducer, initialState)

const MyComponent = ({children}) => {
  return (
    <MyStateProvider>
      <SomeChild>
        <SomeNestedChild />
      </SomeChild>
    </MyStateProvider>
  )
}

const SomeNestedChild = () => {
  const state = useMyState() 
  const dispatch = useMyDispatch()

  // ...
}

```

Reply:
Oh, I kinda like that!